app/ante/ante.go

```diff
@@ -0,0 +1,53 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/crypto-org-chain/cronos/v2/x/cronos/keeper"
+	"github.com/crypto-org-chain/cronos/v2/x/cronos/types"
+	evmante "github.com/evmos/ethermint/app/ante"
+)
+
+// NewAnteHandler add additional logic on top of Ethermint's anteHandler
+func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
+	return func(
+		ctx sdk.Context, tx sdk.Tx, sim bool,
+	) (newCtx sdk.Context, err error) {
+		var anteHandler sdk.AnteHandler
+
+		defer evmante.Recover(ctx.Logger(), &err)
+
+		// Check msg authorization
+		for _, msg := range tx.GetMsgs() {
```

Comment:
Normally these logic should be a decorator, but I guess existing api don't support adding extra decorators.

Reply:
its a bit a hacky way